lumibot/components/agents/builtins.py

```
@@ -459,6 +644,219 @@ def last_price(
+                        price = float(raw) if raw is not None else None
+                    except Exception:
+                        price = None
+                    if price is None or not math.isfinite(price):
+                        prices[symbol] = None
+                        missing.append(symbol)
+                    else:
+                        prices[symbol] = price
+            else:
+                batch_result = None
+        else:
+            batch_result = None
```

Pylint (4.0.6)

[refactor] Too many nested blocks (6/5)

(R1702)